ignite/services/chain/chain.go

```diff
@@ -138,6 +119,14 @@ func EnableThirdPartyModuleCodegen() Option {
 	}
 }
 
+// WithSession sets the CLI output session for the chain.
+func WithSession(s cliui.Session) Option {
+	return func(c *Chain) {
+		c.session = &s
+		c.ev = c.session.EventBus()
```

Comment:
Got it, maybe it would have been better to keep `logLevel` field and to populate it with the `session.Verbosity()` from the caller side ? Then rename `WithSession` to `CollectEvents` like for `service/networks` ?

The concept behind is to not pass ui objects like `Session` there.

I can provide a small patch if you don't mind.

Reply:
Yes, go for it 👍

Reply:
oups OK I missed that the `chaincmdrunner` also needs a stdout and a stderr, which are actually set by the `session` too. So let's keep it like that :pray: :sweat_smile:

Reply:
After thinking how to make it more consistent by using the `CollectEvents` to remove the dependency with the `cliui.Session` from the `Chain` I made these [changes](https://github.com/ignite/cli/pull/2510/commits/fba42d9f1bbbacce9989b64e0f876adcde143b1f). Thanks for pointing it @tbruyelle.

Reply:
nice!